###
### THIS FILE IS AUTOMATICALLY GENERATED by ./gazebo_control_gen.sh /home/k-okada/catkin_ws/ws_nextage/src/rtmros_nextage/nextage_gazebo/config 2016年  1月 27日 水曜日 22:35:15 JST
###
# Publish all joint states -----------------------------------
joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: 50

  # Position Controllers (arms) ---------------------------------------
CHEST_JOINT0_position_controller:
  type: effort_controllers/JointPositionController
  joint: CHEST_JOINT0
  pid: {p: 100, i: 0.01, d: 5.0}
HEAD_JOINT0_position_controller:
  type: effort_controllers/JointPositionController
  joint: HEAD_JOINT0
  pid: {p: 100, i: 0.01, d: 5.0}
HEAD_JOINT1_position_controller:
  type: effort_controllers/JointPositionController
  joint: HEAD_JOINT1
  pid: {p: 100, i: 0.01, d: 5.0}
LARM_JOINT0_position_controller:
  type: effort_controllers/JointPositionController
  joint: LARM_JOINT0
  pid: {p: 100, i: 0.01, d: 5.0}
LARM_JOINT1_position_controller:
  type: effort_controllers/JointPositionController
  joint: LARM_JOINT1
  pid: {p: 100, i: 0.01, d: 5.0}
LARM_JOINT2_position_controller:
  type: effort_controllers/JointPositionController
  joint: LARM_JOINT2
  pid: {p: 100, i: 0.01, d: 5.0}
LARM_JOINT3_position_controller:
  type: effort_controllers/JointPositionController
  joint: LARM_JOINT3
  pid: {p: 100, i: 0.01, d: 5.0}
LARM_JOINT4_position_controller:
  type: effort_controllers/JointPositionController
  joint: LARM_JOINT4
  pid: {p: 100, i: 0.01, d: 5.0}
LARM_JOINT5_position_controller:
  type: effort_controllers/JointPositionController
  joint: LARM_JOINT5
  pid: {p: 100, i: 0.01, d: 5.0}
RARM_JOINT0_position_controller:
  type: effort_controllers/JointPositionController
  joint: RARM_JOINT0
  pid: {p: 100, i: 0.01, d: 5.0}
RARM_JOINT1_position_controller:
  type: effort_controllers/JointPositionController
  joint: RARM_JOINT1
  pid: {p: 100, i: 0.01, d: 5.0}
RARM_JOINT2_position_controller:
  type: effort_controllers/JointPositionController
  joint: RARM_JOINT2
  pid: {p: 100, i: 0.01, d: 5.0}
RARM_JOINT3_position_controller:
  type: effort_controllers/JointPositionController
  joint: RARM_JOINT3
  pid: {p: 100, i: 0.01, d: 5.0}
RARM_JOINT4_position_controller:
  type: effort_controllers/JointPositionController
  joint: RARM_JOINT4
  pid: {p: 100, i: 0.01, d: 5.0}
RARM_JOINT5_position_controller:
  type: effort_controllers/JointPositionController
  joint: RARM_JOINT5
  pid: {p: 100, i: 0.01, d: 5.0}

# Trajectory Controllers ---------------------------------------
larm_controller:
  type: effort_controllers/JointTrajectoryController
  joints:
    - LARM_JOINT0
    - LARM_JOINT1
    - LARM_JOINT2
    - LARM_JOINT3
    - LARM_JOINT4
    - LARM_JOINT5
  constraints:
    goal_time: 0.5                   # Override default
    stopped_velocity_tolerance: 0.50 # Override default
    LARM_JOINT0:
      trajectory: 0.10               # Not enforced if unspecified
      goal: 0.02                     # Not enforced if unspecified
    LARM_JOINT1:
      trajectory: 0.10               # Not enforced if unspecified
      goal: 0.02                     # Not enforced if unspecified
    LARM_JOINT2:
      trajectory: 0.05               # Not enforced if unspecified
      goal: 0.02                     # Not enforced if unspecified
    LARM_JOINT3:
      trajectory: 0.05               # Not enforced if unspecified
      goal: 0.02                     # Not enforced if unspecified
    LARM_JOINT4:
      trajectory: 0.05               # Not enforced if unspecified
      goal: 0.02                     # Not enforced if unspecified
    LARM_JOINT5:
      trajectory: 0.05               # Not enforced if unspecified
      goal: 0.02                     # Not enforced if unspecified

  gains: # Required because we're controlling an effort interface
    LARM_JOINT0: {p: 5000, i: 5, d: 1}
    LARM_JOINT1: {p: 10000, i: 5, d: 1}
    LARM_JOINT2: {p: 5000, i: 5, d: 1}
    LARM_JOINT3: {p: 1500, i: 5, d: 1}
    LARM_JOINT4: {p: 1500, i: 5, d: 1}
    LARM_JOINT5: {p: 1500, i: 5, d: 1}
  state_publish_rate:  100            # Override default
  action_monitor_rate: 100            # Override default
  stop_trajectory_duration: 0        # Override default


# Trajectory Controllers ---------------------------------------
rarm_controller:
  type: effort_controllers/JointTrajectoryController
  joints:
    - RARM_JOINT0
    - RARM_JOINT1
    - RARM_JOINT2
    - RARM_JOINT3
    - RARM_JOINT4
    - RARM_JOINT5
  constraints:
    goal_time: 0.5                   # Override default
    stopped_velocity_tolerance: 0.50 # Override default
    RARM_JOINT0:
      trajectory: 0.10               # Not enforced if unspecified
      goal: 0.02                     # Not enforced if unspecified
    RARM_JOINT1:
      trajectory: 0.10               # Not enforced if unspecified
      goal: 0.02                     # Not enforced if unspecified
    RARM_JOINT2:
      trajectory: 0.05               # Not enforced if unspecified
      goal: 0.02                     # Not enforced if unspecified
    RARM_JOINT3:
      trajectory: 0.05               # Not enforced if unspecified
      goal: 0.02                     # Not enforced if unspecified
    RARM_JOINT4:
      trajectory: 0.05               # Not enforced if unspecified
      goal: 0.02                     # Not enforced if unspecified
    RARM_JOINT5:
      trajectory: 0.05               # Not enforced if unspecified
      goal: 0.02                     # Not enforced if unspecified

  gains: # Required because we're controlling an effort interface
    RARM_JOINT0: {p: 5000, i: 5, d: 1}
    RARM_JOINT1: {p: 10000, i: 5, d: 1}
    RARM_JOINT2: {p: 5000, i: 5, d: 1}
    RARM_JOINT3: {p: 1500, i: 5, d: 1}
    RARM_JOINT4: {p: 1500, i: 5, d: 1}
    RARM_JOINT5: {p: 1500, i: 5, d: 1}
  state_publish_rate:  100            # Override default
  action_monitor_rate: 100            # Override default
  stop_trajectory_duration: 0        # Override default


# Trajectory Controllers ---------------------------------------
head_controller:
  type: effort_controllers/JointTrajectoryController
  joints:
    - HEAD_JOINT0
    - HEAD_JOINT1
  constraints:
    goal_time: 0.5                   # Override default
    stopped_velocity_tolerance: 0.50 # Override default
    HEAD_JOINT0:
      trajectory: 0.05               # Not enforced if unspecified
      goal: 0.02                     # Not enforced if unspecified
    HEAD_JOINT1:
      trajectory: 0.05               # Not enforced if unspecified
      goal: 0.02                     # Not enforced if unspecified

  gains: # Required because we're controlling an effort interface
    HEAD_JOINT0: {p: 1500, i: 5, d: 1}
    HEAD_JOINT1: {p: 1500, i: 5, d: 1}
  state_publish_rate:  100            # Override default
  action_monitor_rate: 100            # Override default
  stop_trajectory_duration: 0        # Override default


# Trajectory Controllers ---------------------------------------
torso_controller:
  type: effort_controllers/JointTrajectoryController
  joints:
    - CHEST_JOINT0
  constraints:
    goal_time: 0.5                   # Override default
    stopped_velocity_tolerance: 0.50 # Override default
    CHEST_JOINT0:
      trajectory: 0.05               # Not enforced if unspecified
      goal: 0.02                     # Not enforced if unspecified

  gains: # Required because we're controlling an effort interface
    CHEST_JOINT0: {p: 1500, i: 5, d: 1}
  state_publish_rate:  100            # Override default
  action_monitor_rate: 100            # Override default
  stop_trajectory_duration: 0        # Override default

